R/get_names.R

```diff
@@ -1,7 +1,15 @@
 # get names according to a pattern. Basically a wrapper around grep ------------
 get_names <- function(pattern, text) {
-  unname(sapply(grep(pattern, text, value = TRUE),
-                function(x) sub("^.*\\\"(.*)\\\".*$", "\\1", x)))
+ # unname(sapply(grep(pattern, text, value = TRUE),
+  #              function(x) sub("^.*\\\"(.*)\\\".*$", "\\1", x)))
+  str <- grep(pattern, text, value = TRUE)
+  lst_str <- lapply(seq_along(str), function(x) {
+    vect <- strsplit(str[x], " var|value")
+    vect <- unlist(vect)[1]
+    vect <- sub("^.*\\\"(.*)\\\".*$", "\\1", vect)
+    vect
+  })
+  unlist(lst_str)
```

Comment:
This function was incorrect?

Reply:
It was missing a little detail. For example, if we were using the function as it was on `NicolasPred.gaml`, for the pattern `parameter`, the result was:
```
"Population parameters", "Population parameters", "Macro parameters", "Macro parameters", "Macro parameters", "Macro parameters", "Micro parameters", "Micro parameters", "Micro parameters", "0", "currentSeed" 
```
Which is not correct. So I add some step to select only the name before`var` or `value` as it is written on the `gaml` for the `monitor` and `parameter` values.

Reply:
Cool! Very nice. It confirms that parsing GAML is really tricky and dangerous.